### PR TITLE
Add AnyBuild pipeline

### PR DESCRIPTION
**Pull request by almiliMSFT**
The pipeline has the following job dependencies:
```
Configure
├── LLVM-linux
├── LLVM-macOS
└── LLVM-windows
```
The `Configure` job sets an output variable named `LLVM_COMMIT` to either to the value of the `$(GOOD_HASH)` pipeline variable (if set) or the commit hash of the llvm-project submodule currently checked into this repo.

The subsequent jobs build the `LLVM_COMMIT` commit of the `llvm-project` repo on Linux, Windows, and macOS.   They all produce an installation tarball and optionally upload it to blob storage.

On Linux and Windows, the build is wrapped with AnyBuild to provide distribution and caching on top of the native build.

### CODE REVIEW

File: devops/llvm-anybuild.yml
Comment (rengolin) [inline devops/llvm-anybuild.yml:14]: Wrong comment. :)

File: devops/llvm-anybuild.yml
Comment (rengolin) [inline devops/llvm-anybuild.yml:58]: Because we're not building Verona in this job anymore, this variable is unnecessary.

File: devops/llvm-anybuild.yml
Comment (rengolin) [inline devops/llvm-anybuild.yml:62]: I'd remove this comment...

File: devops/llvm-anybuild.yml
Comment (rengolin) [inline devops/llvm-anybuild.yml:68]: `VERONA_COMMIT` is now unused. I'd remove all mentions of it.

File: devops/llvm-anybuild.yml
Comment (rengolin) [inline devops/llvm-anybuild.yml:130]: `Choco` and `PowerShell` were used for `CMake` and `ninja` because the image we had didn't have any tools installed, but the standard Windows VM in Azure does. If you look at `ci.yml`, we're using the same image `windows-2019` and you'll see that it's all there and you can use `script` directly.

**Review by rengolin — COMMENTED** [review]
Thanks @almiliMSFT, this looks great! I have a few minor comments only.

Splitting the `yml` files like you did is something we had on our TODO list for a while, thanks for showing the way. :)

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml]: Updated

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml]: deleted VERONA_COMMIT variable

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml]: removed

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml]: removed

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml:124]: removed choco and ninja installation steps

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [reply devops/llvm-anybuild.yml]: I left a similar comment next to the `Configure` job though.

**Comment by almiliMSFT** [comment]
Running a test build [here](https://dev.azure.com/ProjectVeronaCI/Project%20Verona/_build/results?buildId=2453&view=results)

File: devops/llvm-anybuild.yml
Comment (almiliMSFT) [inline devops/llvm-anybuild.yml:1]: Running a test build [here](https://dev.azure.com/ProjectVeronaCI/Project%20Verona/_build/results?buildId=2453&view=results)

Comment (almiliMSFT) [reply devops/llvm-anybuild.yml:1]: And another one [here](https://dev.azure.com/ProjectVeronaCI/Project%20Verona/_build/results?buildId=2454&view=results)

**Review by rengolin — APPROVED** [review]
LGTM, thanks! Is it ready to merge?

**Comment by almiliMSFT** [comment]
Ready to merge.

One thing to be improved at some point is cleaning up the environment variables further before executing AnyBuild on Windows.  A bunch of irrelevant OS env variables may change between runs (e.g., when the VM image gets update `IMAGEVERSION`, `GOROOT`, etc. all change) causing unnecessary cache misses.  

On Linux the build runs in a clean environment (e.g., created with `env -i ...`) so this problem does not occur there.  I don't know how to do the same on Windows.

**Comment by rengolin** [comment]
A completely unrelated error occurred. It could have been due to a recent change in the runtime. I'm running the tests again, let's see if it was due to instability.

**Comment by rengolin** [comment]
Weird, the error is nor spurious, ASAN is definitely not liking the tests. But running the CI on the master branch isn't showing anything because we don't detect changes and don't build or test anything. :(

I'll look more into it and see what's going on.

**Comment by rengolin** [comment]
Ok, something is seriously wrong with our tests and has nothing to do with this PR. 

But, as it stands, the tests are broken on master and we have a Github rule not to merge with breaking tests, so we'll have to wait until the tests are all passing on master to merge this PR.

**Pull request merged**
